README.md

```diff
@@ -46,7 +249,48 @@ async () => {
 }
 ```
 
-### Consuming messages with consumer groups
+#### <a name="producing-messages-custom-partitioner"></a> Custom partitioner
+
+It's possible to assign a custom partitioner to the consumer. A partitioner is a function which returns another function responsible for the partition selection, something like this:
+
+```javascript
+const MyPartitioner = () => {
+  // some initialization
+  return ({ topic, partitionMetadata, message }) => {
+    // select a partition based on some logic
+    // return the partition number
+    return 0
+  }
+}
+```
+
+`partitionMetadata` is an array of partitions with the following structure:
+
+```javascript
+[
+  { partitionId: 1, leader: 1 },
+  { partitionId: 2, leader: 2 },
+  { partitionId: 0, leader: 0 }
+]
+```
+
+To Configure your partitioner use the option `createPartitioner`.
+
+```javascript
+kafka.producer({ createPartitioner: MyPartitioner })
+```
+
+#### <a name="producing-messages-retry"></a> Retry
+
+The option `retry` can be used to customize the configurations for the producer.
```

Comment:
configuration: singular not configuration**s**

Reply:
done